lib/cisco_node_utils/interface.rb

```diff
@@ -420,6 +420,108 @@ def default_ipv4_arp_timeout
       config_get_default('interface', ipv4_arp_timeout_lookup_string)
     end
 
+    def ipv4_dhcp_relay_addr
+      config_get('interface', 'ipv4_dhcp_relay_addr', name: @name)
+    end
+
+    def ipv4_dhcp_relay_addr=(list)
+      cur_list = ipv4_dhcp_relay_addr
+      # remove the current addresses first
+      unless cur_list.empty?
+        cur_list.each do |addr|
+          config_set('interface', 'ipv4_dhcp_relay_addr',
+                     name:  @name, state: 'no', addr: addr)
+        end
+      end
+      Feature.dhcp_enable unless list.empty?
+      list.each do |addr|
+        config_set('interface', 'ipv4_dhcp_relay_addr',
+                   name:  @name, state: '', addr: addr)
+      end
+    end
+
+    def default_ipv4_dhcp_relay_addr
+      config_get_default('interface', 'ipv4_dhcp_relay_addr')
+    end
+
+    def ipv4_dhcp_relay_info_trust
+      config_get('interface', 'ipv4_dhcp_relay_info_trust', name: @name)
+    end
+
+    def ipv4_dhcp_relay_info_trust=(state)
+      Feature.dhcp_enable if state
+      config_set('interface', 'ipv4_dhcp_relay_info_trust',
+                 name:  @name, state: state ? '' : 'no')
+    end
+
+    def default_ipv4_dhcp_relay_info_trust
+      config_get_default('interface', 'ipv4_dhcp_relay_info_trust')
+    end
+
+    def ipv4_dhcp_relay_src_addr_hsrp
+      config_get('interface', 'ipv4_dhcp_relay_src_addr_hsrp', name: @name)
+    end
+
+    def ipv4_dhcp_relay_src_addr_hsrp=(state)
+      Feature.dhcp_enable if state
+      config_set('interface', 'ipv4_dhcp_relay_src_addr_hsrp',
+                 name:  @name, state: state ? '' : 'no')
+    end
+
+    def default_ipv4_dhcp_relay_src_addr_hsrp
+      config_get_default('interface', 'ipv4_dhcp_relay_src_addr_hsrp')
+    end
+
+    # manifest will have 'ethernet 1/1', port-channel 100', vlan 2' etc.
+    # The show cmd returns 'Ethernet1/1', 'port-channel100', Vlan2, etc.
+    # so make this idempotent, normalize the output to match the input.
+    def ipv4_dhcp_relay_src_intf
```

Comment:
If I understand correctly you are trying to downcase and insert whitespace?  I suggest you just normalize by downcasing and removing whitespace.

```
irb(main):012:0* intf = "Ethernet 1/1"
=> "Ethernet 1/1"
irb(main):013:0> intf.downcase.delete(' ')
=> "ethernet1/1"
irb(main):014:0> intf = "Ethernet1/1"
=> "Ethernet1/1"
irb(main):015:0> intf.downcase.delete(' ')
=> "ethernet1/1"
irb(main):016:0> intf = "ethernet 1/1"
=> "ethernet 1/1"
irb(main):017:0> intf.downcase.delete(' ')
=> "ethernet1/1"
irb(main):018:0> 
```
You can do this here and munge it in the type file so users can enter the name in any of these formats.